src/derivable/data-source.ts

```diff
@@ -0,0 +1,223 @@
+import { isRecordingObservations, recordObservation } from '../tracking';
+import { processChangedAtom } from '../transaction';
+import { debugMode, equals, MixinFn, MixinProp } from '../utils';
+import { Atom } from './atom';
+import { Derivable } from './derivable';
+import { Lens } from './lens';
+import { AtomPluck, pluck } from './pluck';
+
+export const EMPTY_CACHE = {};
+
+export abstract class DataSource<V> extends Derivable<V> implements Atom<V> {
```

Comment:
Misschien is het een idee om hier een functie (zoals `atom()` etc) omheen te bouwen. Ik kan me voorstellen dat je voor simpele `DataSource`s, niet altijd een `superclass` aan wil maken. 

Zeg je hebt een ticker oid, hoef je in theorie alleen enkele functies aan te leveren die hun ding doen en that's it..